yt_dlp/extractor/blogger.py

```diff
@@ -0,0 +1,49 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from ..utils import mimetype2ext, parse_duration, parse_qs, str_or_none, traverse_obj
+from .common import InfoExtractor
+
+
+class BloggerIE(InfoExtractor):
+    IE_NAME = 'blogger.com'
+    _VALID_URL = r'https?://(?:www\.)?blogger\.com/video\.g\?token=(?P<id>.+)'
+    _VALID_EMBED = r'''<iframe[^>]+src=["']((?:https?:)?//(?:www\.)?blogger\.com/video\.g\?token=[^"']+)["']'''
+    _TESTS = [{
+        'url': 'https://www.blogger.com/video.g?token=AD6v5dzEe9hfcARr5Hlq1WTkYy6t-fXH3BBahVhGvVHe5szdEUBEloSEDSTA8-b111089KbfWuBvTN7fnbxMtymsHhXAXwVvyzHH4Qch2cfLQdGxKQrrEuFpC1amSl_9GuLWODjPgw',
+        'md5': 'f1bc19b6ea1b0fd1d81e84ca9ec467ac',
+        'info_dict': {
+            'id': 'BLOGGER-video-3c740e3a49197e16-796',
+            'title': 'BLOGGER-video-3c740e3a49197e16-796',
+            'ext': 'mp4',
+            'thumbnail': r're:^https?://.*',
+        }
+    }]
+
+    @staticmethod
+    def _extract_urls(webpage):
+        return re.findall(BloggerIE._VALID_EMBED, webpage)
+
+    def _real_extract(self, url):
+        token_id = self._match_id(url)
+        webpage = self._download_webpage(url, token_id)
+        data_json = self._search_regex(r'var\s+VIDEO_CONFIG\s*=\s*(\{.*)', webpage, 'JSON data')
+        data_json = data_json.encode('utf-8').decode('unicode_escape')
+        data = self._parse_json(data_json, token_id)
+        streams = data['streams']
+        formats = [{
+            'ext': mimetype2ext(traverse_obj(parse_qs(stream['play_url']), ('mime', 0))),
+            'url': stream['play_url'],
+            'format_id': str_or_none(stream.get('format_id')),
+            'duration': parse_duration(traverse_obj(parse_qs(stream['play_url']), ('dur', 0))),
```

Comment:
`duration` is not a permitted field for individual formats atm

Reply:
```suggestion
```

Reply:
In practice it seems they don't, although in theory they could. The main reason I added this is for the duration in the dict returned from the function to be able to just get the first duration:

```
            'duration': streams[0].get('duration'),
```

Reply:
wdym? you are not taking the global duration from inside the formats dict

Reply:
Thats a bug, it should be like this:

```
            'duration': formats[0].get('duration'),
```

Reply:
why not directly
```py
parse_duration(traverse_obj(parse_qs(streams[0]['play_url']), ('dur', 0)))
```

Reply:
I had that initially, but the other way felt more correct. I'll revert it back the other way.

Reply:
Done.